plugins/workspace-search/src/workspace_search.ts

```diff
@@ -132,9 +132,10 @@ export class WorkspaceSearch implements Blockly.IPositionable {
      * </div>
      */
     const injectionDiv = this.workspace.getInjectionDiv();
-    this.addEvent(injectionDiv, 'keydown', this, (evt: KeyboardEvent) =>
-      this.onWorkspaceKeyDown(evt),
-    );
+    this.addEvent(injectionDiv, 'keydown', this, (evt: Event) => {
+      const keyboardEvent = evt as KeyboardEvent; 
```

Comment:
Instead of casting all of these individually, I think you can make `addEvent` generic and have this take the correct event from the get-go.

```ts
private addEvent<T extends Event>(
    node: Element,
    name: string,
    thisObject: object,
    func: (event: T) => void,
  ) { }
```

Reply:
Hello! For this one, instead of adding a new local function, can you modify the method on the WorkspaceSearch class to be generic?

Reply:
<img width="1130" alt="Screenshot 2023-11-01 at 21 51 56" src="https://github.com/google/blockly-samples/assets/91741581/14f111c0-0d77-42c0-8227-8ded4e6135f5">
So inside this class, shall I create a **`addEvent`** generic ?

Reply:
It already exists you ust need to modify it: https://github.com/google/blockly-samples/blob/39037d1e2113174c9d60c1f9389eddbe7df589e3/plugins/workspace-search/src/workspace_search.ts#L196

Reply:
private addEvent <T extends Event>(
    node: Element,
    name: string,
    thisObject: object,
    func: (event: T) => void,
  ) {
    const event = Blockly.browserEvents.conditionalBind(
      node,
      name,
      thisObject,
      func,
    );
    this.boundEvents.push(event);
  }
  I've modified. Is this alright?

Reply:
Looks correct! Can you change the callers to use the type they expect? And then remove the casts you added?

Reply:
Do I need to assign types to node, name, thisObject?

Reply:
> Looks correct! Can you change the callers to use the type they expect? And then remove the casts you added?

Can you please explain this one a bit?

Reply:
Yeah! When you're calling `this.addEvent` you should pass a callback that says the type it expects. E.g. `this.addEvent(injectionDiv, 'keydown', this, (evt: KeyboardEvent) => {})`

Reply:
> Yeah! When you're calling `this.addEvent` you should pass a callback that says the type it expects. E.g. `this.addEvent(injectionDiv, 'keydown', this, (evt: KeyboardEvent) => {})`

So shall I replace all the callers with injectionDiv whereever this.Element is used??

Reply:
Nope, we're trying to fix the type of the event in the callback. I changed it to use `KeyboardEvent` instead of `Event`!

Reply:
> KeyboardEvent

So what to do now?

Reply:
```
this.addEvent(this.inputElement, 'keydown', this, (evt: Event) => {
  const keyboardEvent = evt as KeyboardEvent;
  this.onKeyDown(keyboardEvent);
});
```

Becomes
```
this.addEvent(this.inputElement, 'keydown', this, (evt: KeyboardEvent) => {
  this.onKeyDown(evt);
});
```

We are now passing a callback that says the type it expects.

Please update for all calls to `this.addEvent`.